addons/auth_signup/models/res_partner.py

```diff
@@ -45,6 +46,20 @@ def _compute_signup_url(self):
                 self.env['res.users'].check_access_rights('write')
             partner.signup_url = result.get(partner.id, False)
 
+    @api.depends('signup_token')
+    def _compute_redirect_url(self):
+        for partner in self:
+            partner_sudo = partner.sudo()
```

Comment:
Wondering about this sudo ... I see there are some other sudo in code. And this is limited to partner without users. But a read access allow to prepare a signup token to any partner... @xmo-odoo do you know a bit flows involving auth signup ? Isn't it a bit touchy ?

Reply:
>  @xmo-odoo do you know a bit flows involving auth signup ? 

nope

> Isn't it a bit touchy ?

i'd expect it to be yes.

Reply:
> do you know a bit flows involving auth signup ? 

Not really.

> Isn't it a bit touchy ?

Yes it seems odd. The existing code does require (doesn't bypass) write access to the partner for the compute. It does, however, sudo the `_get_signup_url_for_action`, which itself sudoes `signup_prepare` internally. I don't think setting a compute is ACL-checked, so in effect the existing one also bypasses ACLs for generation (the only thing it checks is a write access to users *if* trying to generate a signup URL for a partner with an internal user).